compiler/vm.nim

```diff
@@ -2185,29 +2185,36 @@ proc evalStaticStmt*(module: PSym; g: ModuleGraph; e: PNode, prc: PSym) =
 proc setupCompileTimeVar*(module: PSym; g: ModuleGraph; n: PNode) =
   discard evalConstExprAux(module, g, nil, n, emStaticStmt)
 
-proc prepareVMValue(arg: PNode): PNode =
+proc prepareVMValue(g: ModuleGraph, arg: PNode): PNode =
   ## strip nkExprColonExpr from tuple values recurively. That is how
   ## they are expected to be stored in the VM.
 
   # Early abort without copy. No transformation takes place.
   if arg.kind in nkLiterals:
     return arg
 
+  if arg.kind == nkExprColonExpr and arg[0].typ != nil and
+     arg[0].typ.sym != nil and $arg[0].typ == "NimNode" and
+     arg[0].typ.owner == g.systemModule:
```

Comment:
Is there a better solution for this gross hack?

Reply:
@Araq Perhaps a new node kind that wraps around NimNode, in oder to treat static[NimNode] as a value and not AST.
I suppose we could also forbid static[NimNode], but that's a breaking change (although using static[NimNode] is not a good idea rn, regardless of this change)

Reply:
this should work:
```nim
  if arg.kind == nkExprColonExpr and arg[0].typ != nil and arg[0].typ.sym != nil and
    arg[0].typ.sym.magic == mPNimrodNode:
```

or simpler, with `import std/wrapnils` now that it doesn't depend on dotoperators:
```nim
if arg.kind == nkExprColonExpr and ?.arg[0].typ.sym.magic == mPNimrodNode:
```

That said, can this branch ever be exercised? which code generates a `static NimNode`?

Reply:
Applied timothees suggestion, should be slightly less of a hack now (it still is one, but it's one to support the static NimNode hack :).

Reply:
but when can a static NimNode occur? at least add a comment mentioning it's not clear if this branch can ever be taken

Reply:
also @Clyybber your commit (which you then reverted) used:
```
 if arg.kind == nkExprColonExpr and arg[0].typ != nil and
     arg[0].typ.sym != nil and arg[0].sym.magic == mPNimrodNode:
```

instead of
```
  if arg.kind == nkExprColonExpr and arg[0].typ != nil and arg[0].typ.sym != nil and
    arg[0].typ.sym.magic == mPNimrodNode
```
which i had suggested above and should work

Reply:
oh haha, nice catch, I think that's it